diesel/src/pg/connection/mod.rs

```diff
@@ -652,24 +654,31 @@ mod tests {
         use crate::sql_query;
 
         let conn = &mut connection();
-        sql_query("LISTEN test_notifications").execute(conn).unwrap();
-        sql_query("NOTIFY test_notifications, 'first'").execute(conn).unwrap();
-        sql_query("NOTIFY test_notifications, 'second'").execute(conn).unwrap();
-        let mut iter = conn.notifications_iter();
-        let first = iter.next().unwrap();
-        let second = iter.next().unwrap();
-        let third = iter.next();
+        sql_query("LISTEN test_notifications")
+            .execute(conn)
+            .unwrap();
+        sql_query("NOTIFY test_notifications, 'first'")
+            .execute(conn)
+            .unwrap();
+        sql_query("NOTIFY test_notifications, 'second'")
+            .execute(conn)
+            .unwrap();
+
+        let (first, second, third) = {
+            let mut iter = conn.notifications_iter();
+            (iter.next().unwrap(), iter.next().unwrap(), iter.next())
+        };
```

Comment:
Can we use `let mut notifications = conn.notifications_iter().collect::<Vec<_>>()` here instead and check that this vector contains exactly two elements. We could then try to fetch the next notification via `conn.notifications_iter().next()` to check that there is no notification.